### ISSUE TITLE
DELETE - DELETED ROWS - ACCIDENTAL

### ISSUE THREAD

**Issue by DingBEN** [pawelsalawa/sqlitestudio#3639]
### Details
1. A DELETE in a visual interface should always ask for confirmation, by DEFAULT.
I just go done typing in a row, and hit the enter key so that I can pan across the row to make sure all is well... I inadvertently grazed the Del key... POOOF! all gone!? (dammit!)
think!
- the Del key on keyboard is exactly **NEXT to the Enter key**
- 'delete button' clicked by mistake (**so similar to the rollback button**)
- etc
If some users prefer to live dangerously, provide a setting in config to by-pass confirmation 

And for that matter, is there a recovery functionality in SQLiteStudio to reverse an accidental delete?
...before a VACUUM is performed!?

SEE RELATED ISSUE: COLORS in SQLiteStudio and (VISUAL)ERGONOMICS

**Comment by pawelsalawa** [comment]
Right. For newly entered & uncommited row this is a problem. Thanks! This looks quite critical, so I'll put it to 3.2.2 bucket.

**Comment by DingBEN** [comment]
any way to get to lost data? even if a hack...

**Comment by pawelsalawa** [comment]
I've done it once. It's not easy, but doable (I read SQLite low level tech docs, analyzed binary data of the file and wrote a script to extract data).

It's possible to implement SQLiteStudio plugin for that, but bare in mind, that if you delete data, it's marked as "deleted", not really deleted and it's like with file systems - you can recover it up until it gets overwritten by new data.

With SQLite it's exceptionally difficult task, as in this database every single cell can have different data type, different representation & different data length. Column types are just a hint, not an enforced definition. In strongly typed databases you can assume datatype (thus it's size & representation in binary format). Not the case in SQLite.

**Comment by DingBEN** [comment]
On a post on stackoverflow, one individual who appeared advanced, stated that the data is immediately deleted. Was a disappointment for me, new to SQLite.
You set it straight... just like the OS! GOOD
I did recover my data using an HEX Editor. LOL sucky, but worked.
